apps/backend/src/lib/emails.tsx

```diff
@@ -63,62 +62,152 @@ export type EmailConfig = {
   type: 'shared' | 'standard',
 }
 
+type SendEmailOptions = {
+  emailConfig: EmailConfig,
+  to: string | string[],
+  subject: string,
+  html?: string,
+  text?: string,
+}
+
+export async function sendEmailWithKnownErrorTypes(options: SendEmailOptions): Promise<Result<undefined, {
+  rawError: any,
+  errorType: 'UNKNOWN' | 'HOST_NOT_FOUND' | 'AUTH_FAILED' | 'SOCKET_CLOSED' | 'TEMPORARY' | 'INVALID_EMAIL_ADDRESS',
+  retryable: boolean,
+  message?: string,
+}>> {
+  try {
+    const transporter = nodemailer.createTransport({
+      host: options.emailConfig.host,
+      port: options.emailConfig.port,
+      secure: options.emailConfig.secure,
+      auth: {
+        user: options.emailConfig.username,
+        pass: options.emailConfig.password,
+      },
+    });
+
+    await transporter.sendMail({
+      from: `"${options.emailConfig.senderName}" <${options.emailConfig.senderEmail}>`,
+      ...options,
+    });
+
+    return Result.ok(undefined);
+  } catch (error) {
+    if (error instanceof Error) {
+      const code = (error as any).code as string | undefined;
+      const responseCode = (error as any).responseCode as number | undefined;
+      const errorNumber = (error as any).errno as number | undefined;
+
+      const getServerResponse = (error: any) => {
+        if (error.response) {
+          return `\nResponse from the email server:\n${error.response}`;
+        }
+        return '';
+      };
+
+      if (errorNumber === -3008 || code === 'EDNS') {
+        return Result.error({
+          rawError: error,
+          errorType: 'HOST_NOT_FOUND',
+          retryable: false,
+          message: 'Failed to connect to the email host. Please make sure the email host configuration is correct.'
+        });
+      }
+
+      if (responseCode === 535 || code === 'EAUTH') {
+        return Result.error({
+          rawError: error,
+          errorType: 'AUTH_FAILED',
+          retryable: false,
+          message: 'Failed to authenticate with the email server. Please check your email credentials configuration.',
+        });
+      }
+
+      if (responseCode === 450) {
+        return Result.error({
+          rawError: error,
+          errorType: 'TEMPORARY',
+          retryable: true,
+          message: 'The email server returned a temporary error. This could be due to a temporary network issue or a temporary block on the email server. Please try again later.' + getServerResponse(error),
```

Comment:
```suggestion
          message: 'The email server returned a temporary error. This could be due to a temporary network issue or a temporary block on the email server. Please try again later.\n\nError: ' + getServerResponse(error),
```